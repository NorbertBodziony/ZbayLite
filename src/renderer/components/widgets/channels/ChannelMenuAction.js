import React from 'react'
import PropTypes from 'prop-types'
import * as R from 'ramda'

import { withStyles } from '@material-ui/core/styles'

import MoreHorizIcon from '@material-ui/icons/MoreHoriz'

import IconButton from '../../ui/IconButton'
import MenuAction from '../../ui/MenuAction'
import MenuActionItem from '../../ui/MenuActionItem'

const styles = theme => ({
  menuItem: {
    height: 8,
    width: 80,
    paddingLeft: theme.spacing(2.5),
    paddingRight: theme.spacing(2.5),
    fontSize: '0.75rem'
  },
  menuList: {
    padding: `${theme.spacing(1.5)}px 0`
  }
})

export const ChannelMenuAction = ({ classes, onInfo, onMute, onDelete }) => (
  <MenuAction
    classes={{
      button: classes.button,
      icon: classes.icon
    }}
    Icon={MoreHorizIcon}
    IconButton={IconButton}
    offset='0 8'
  >
    <MenuActionItem onClick={onInfo} title='Info' />
    <MenuActionItem onClick={onMute} title='Mute' />
    <MenuActionItem onClick={onDelete} title='Delete' />
  </MenuAction>
)

ChannelMenuAction.propTypes = {
  classes: PropTypes.object.isRequired,
  onInfo: PropTypes.func.isRequired,
  onMute: PropTypes.func.isRequired,
  onDelete: PropTypes.func.isRequired
}

export default R.compose(
  React.memo,
  withStyles(styles)
)(ChannelMenuAction)
