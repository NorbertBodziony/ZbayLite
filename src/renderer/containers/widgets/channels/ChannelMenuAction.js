import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import ChannelMenuAction from '../../../components/widgets/channels/ChannelMenuAction'

export const mapDispatchToProps = dispatch => bindActionCreators({
  onInfo: () => console.warn('[ChannelMenuAction] onInfo not implemented'),
  onMute: () => console.warn('[ChannelMenuAction] onMute not implemented'),
  onDelete: () => console.warn('[ChannelMenuAction] onDelete not implemented')
}, dispatch)

export default connect(null, mapDispatchToProps)(ChannelMenuAction)
