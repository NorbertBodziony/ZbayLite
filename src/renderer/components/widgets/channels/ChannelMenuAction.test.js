import React from 'react'
import { shallow } from 'enzyme'

import { mockClasses } from '../../../../shared/testing/mocks'
import { ChannelMenuAction } from './ChannelMenuAction'

describe('ChannelMenuAction', () => {
  it('renders component', () => {
    const result = shallow(
      <ChannelMenuAction
        classes={mockClasses}
        onInfo={jest.fn()}
        onMute={jest.fn()}
        onDelete={jest.fn()}
      />
    )
    expect(result).toMatchSnapshot()
  })
})
