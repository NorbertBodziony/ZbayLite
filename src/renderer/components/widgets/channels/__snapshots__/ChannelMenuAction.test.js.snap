// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelMenuAction renders component 1`] = `
<WithStyles(MenuAction)
  Icon={[Function]}
  IconButton={
    Object {
      "$$typeof": Symbol(react.memo),
      "compare": null,
      "type": [Function],
    }
  }
  classes={
    Object {
      "button": "button-mock",
      "icon": "icon-mock",
    }
  }
  menuItems={
    Array [
      Immutable.MenuItem {
        "title": "Info",
        "onClick": [MockFunction],
      },
      Immutable.MenuItem {
        "title": "Mute",
        "onClick": [MockFunction],
      },
      Immutable.MenuItem {
        "title": "Delete",
        "onClick": [MockFunction],
      },
    ]
  }
  offset="0 8"
/>
`;
